packages/gatsby/src/redux/reducers/html.ts

```diff
@@ -0,0 +1,123 @@
+import { ActionsUnion, IGatsbyState, IHtmlFileState } from "../types"
+
+const FLAG_DIRTY_CLEARED_CACHE = 0b0000001
+const FLAG_DIRTY_NEW_PAGE = 0b0000010
+const FLAG_DIRTY_PAGE_QUERY_CHANGED = 0b0000100 // TODO: this need to be PAGE_DATA and not PAGE_QUERY, but requires some shuffling
```

Comment:
👍 Also noticed that `PAGE_QUERY` is a bit confusing name for what it actually is.

Reply:
https://github.com/gatsbyjs/gatsby/pull/29498 is one that addresses TODO here. This PR is largely 1 to 1 refactor of current stuff and introduction of `html` reducer and tried to avoid making more changes than just to keep what we currently track so far